src/controllers/userController.js

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable require-jsdoc */
+import helpers from '../utils/helper';
+import stringHelpers from '../utils/stringHelper';
+import db from '../database/models/index';
+import sendResult from '../utils/sendResult';
+
+const User = {
+  async signup(req, res) {
+    const { username, password, email } = req.body;
+    const existingUser = await db.Users.findOne({ where: { email }, raw: true });
```

Comment:
@lemoissonM  am suggesting if you make get `single user by email` a middleware so that we can reuse it  to avoid repetition in creating the same function